resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/Client.scala

```diff
@@ -73,6 +76,28 @@ private[spark] class Client(
   private val driverJavaOptions = sparkConf.get(
     org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
 
+  // create resource of kind - SparkJob representing the deployed spark app
+  private val sparkJobCtrller = new SparkJobResourceClientFromOutsideK8s(
+    kubernetesClientProvider.get)
+  private val jobResourceName = s"sparkJob-${sparkConf.get(KUBERNETES_NAMESPACE)}-$appName"
+  private val keyValuePairs = Map(
+    "image" -> driverDockerImage,
+    "state" -> JobState.QUEUED,
+    "numExecutors" -> sparkConf.getInt("spark.executor.instances", 1),
+    "sparkDriver" -> kubernetesDriverPodName
+  )
+
+  // Failure might be due to TPR inexistence or maybe we're stuck in the 10 minute lag
+  // TODO: in the latter case we can attempt a retry depending on the rc
+  // This also assumes that once we fail at creation, we won't bother trying
+  // anything on the resource for the lifetime of the app
+  Try(sparkJobCtrller.createJobObject(jobResourceName, keyValuePairs)) match {
```

Comment:
We usually use

    try {
      ...
    } catch {
      case e: SparkException =>
    }

I don't think the Spark codebase uses Scala's `Try(...) match` idiom. However I actually think the `Try(...) match` idiom makes sense here, so if there's other examples of this being done in the code then it's also appropriate to do so here.